constants/role.constant.js

```diff
@@ -151,27 +151,26 @@ function createAllSingularRoles() {
     const allRoutes = Constants.Routes.allRoutes;
     let roles = [];
 
-    // i is unique integer so that objectId is constant
-    var i = 1000000;
     for (let routeGroupKey in allRoutes) {
-        if (!allRoutes.hasOwnProperty(routeGroupKey)) {
+        if (!Object.prototype.hasOwnProperty.call(allRoutes, routeGroupKey)) {
             continue;
         }
 
         const routeGroup = allRoutes[routeGroupKey];
         for (let routeKey in routeGroup) {
-            if (!routeGroup.hasOwnProperty(routeKey)) {
+            // Iterating through the attributes in the routeGroup object
+            if (!Object.prototype.hasOwnProperty.call(routeGroup, routeKey)) {
+                // Avoid all prototype attributes
                 continue;
             }
 
             let role = {
-                _id: mongoose.Types.ObjectId(i),
+                _id: routeGroup[routeKey]._id,
```

Comment:
Maybe it would be cleaner to put `_id: mongoose.Types.ObjectId.createFromTime(routeGroup[routeKey]._id)` here and make routes' `_id`s just be integers? On the other hand, having an `_id` that wasn't an ObjectId could be confusing.

Reply:
The benefit of keeping the objectId definition of `_id` in `routes.constant.js` is that we are now able to automatically include that attribute in the list of `route` objects in the `user roles`. See this example: 

<img width="814" alt="Screen Shot 2020-07-26 at 4 19 12 PM" src="https://user-images.githubusercontent.com/17463268/88488621-5f3d1180-cf5c-11ea-9081-dfc88c5200b2.png">

The referent document:
![image](https://user-images.githubusercontent.com/17463268/88488630-7a0f8600-cf5c-11ea-8e5c-ff7c8023e461.png)

Reply:
Ah gotcha